src/cbmc/cbmc_parse_options.cpp

```diff
@@ -553,6 +546,26 @@ int cbmc_parse_optionst::doit()
   if(set_properties())
     return CPROVER_EXIT_SET_PROPERTIES_FAILED;
 
+  // At this point, our goto-model should be in core-goto form (all of the
+  // transformations have been run and the program is ready to be given to the
+  // solver).
+  if(cmdline.isset("export-core-goto"))
```

Comment:
:pick: I think that ideally the contents of `cmdline` should be validated and written to `options` in the `cbmc_parse_optionst::get_command_line_options` function, before the options are acted on at this point in the control flow. This helps separate the validation of the command line arguments supplied by the user from the code which acts on the options that are specified by those arguments. Think of `cmdline` as being the parse tree and `options` as being the setting being acted upon.

Reply:
Done.